docs/reference/changelog.rst

```diff
@@ -13,6 +13,8 @@ Application
 - Add :py:attr:`~craft_application.application.AppMetadata.artifact_type` to
   ``AppMetadata`` to allow user-visible messages to refer to the app-specific output
   artifact type.
+- When packing with ``--debug``, the offending error message will now be displayed
+  at the beginning of the shell session.
```

Comment:
I really like what you wrote in the PR description. :) Could we include the bit about it being printed twice? I think that would bring us closer to the truth.